### PR TITLE
Add unsigned transact function

### PR DESCRIPTION
**Pull request by gakonst**
- Return unsigned transaction if there's no signer attached
- Add UnsignedTransact binding, so that we can just get the tx

### CODE REVIEW

File: accounts/abi/bind/base.go [hunk 1]
```diff
@@ -232,7 +289,7 @@ func (c *BoundContract) transact(opts *TransactOpts, contract *common.Address, i
 		rawTx = types.NewTransaction(nonce, c.address, value, gasLimit, gasPrice, input)
 	}
 	if opts.Signer == nil {
-		return nil, errors.New("no signer to authorize the transaction with")
+		return rawTx, errors.New("no signer to authorize the transaction with")
```
Comment (enlight) [inline accounts/abi/bind/base.go]: Returning a valid value while also returning an error violates errors handling conventions. Why is this change needed?

**Review by enlight — CHANGES_REQUESTED** [review]

File: accounts/abi/bind/base.go (hunk 1 above)
Comment (gakonst) [reply accounts/abi/bind/base.go]: Ah forgot to unstage this change, nvm.

**Review by enlight — APPROVED** [review]
LGTM
